airflow/jobs.py

```diff
@@ -191,11 +198,16 @@ def run(self):
         self.id = id_
 
         # Run
-        self._execute()
+        try:
+            self._execute()
+            self.state = State.SUCCESS
+        except Exception as e:
+            self.state = State.FAILED
+            logging.exception(e)
+            self.on_failure(e)
 
         # Marking the success in the DB
```

Comment:
This comment should be updated to indicate that we are saving the state (not just success)